batch/kvdb.go

```diff
@@ -0,0 +1,35 @@
+package batch
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/lightningnetwork/lnd/sqldb"
+)
+
+// BoltBackend is a bbolt implementation of the sqldb.BatchedTx interface.
+type BoltBackend[Q kvdb.RwTx] struct {
+	db kvdb.Backend
+}
+
+// NewBoltBackend creates a new BoltBackend instance.
+func NewBoltBackend[Q kvdb.RwTx](db kvdb.Backend) *BoltBackend[Q] {
+	return &BoltBackend[Q]{db: db}
+}
+
+// ExecTx will execute the passed txBody, operating upon generic
+// parameter Q (usually a storage interface) in a single transaction.
+//
+// NOTE: This is part of the sqldb.BatchedTx interface.
+func (t *BoltBackend[Q]) ExecTx(_ context.Context, opts sqldb.TxOptions,
+	txBody func(Q) error, reset func()) error {
+
+	if opts.ReadOnly() {
+		return fmt.Errorf("read-only transactions not supported")
```

Comment:
nit: why not?

Reply:
cause of the `kvdb.Update` call below - just trying to keep things identical to how it was before making it generic - ie, only `kvdb.Update` is used. Technically, the `sqldb.TxOptions` here is only used by the SQL impl so guess we can just ignore it here. Was just a safety thing. 
could probs remove the check - but yeah, worth noting that we only use the batcher for write txs at the moment

Reply:
Yes, agree with your intention but was wondering if we should take the step and just add it? Could be quite useful to generalize the scheduler request to `Do(tx Q)` something instead of `Update` and here based on the tx option just call `View` or `Update` for the kv backend. Also it might be worth considering moving `BatchedTx` and `TxOptions` out of `sqldb` so they'd no longer be tightly coupled with SQL conceptually.

Either way, just a food for thought.

Reply:
ok cool: to keep this commit a pure refactor, i've left it as is. Then i've added on a commit at the end of the PR that does as suggested. 

Let me know what you think - i think if it requires many more iterations then perhaps worth it to leave to another PR just so that we dont get too side tracked here

Reply:
lgtm!!